misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/OptionalAssertionsSpec.kt

```diff
@@ -10,8 +10,9 @@ import org.spekframework.spek2.style.specification.Suite
 import java.lang.AssertionError
 import java.util.Optional
 
-abstract class OptionalAssertionsSpec (
+abstract class OptionalAssertionsSpec(
     isEmpty: Fun0<Optional<Any>>,
+    isPresent: Fun1<Optional<Any>, Any>,
```

Comment:
That's probably one of the reasons why your spec does not compile. You want a `Feature0` and not a `Fun1`. This is Atrium specific and thus a little explanation, `FunX` means simplified that you want to test an assertion function which expects X arguments (`Fun1` expects one argument). Your function expects 0, so that's one mistake. Moreover, a `FunX` has the same return type as receiver type (`Optional<Any>` in your example). That's wrong for your case as well, your return type changes to `Any`. Therefore you need to use Feature0 => see ResultFeatureAssertionsSpec

Moreover, you should rename isPresent to `isPresentFeature` (corresponds to `isSuccessFeature` in ResultFeatureAssertionsSpec) and `isPresent` analogous to `isSuccess` in ResultFeatureAssertionsSpec